test/fuzz/mmap_alloc.cc

```diff
@@ -0,0 +1,29 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2021-2022 Intel Corporation
+ */
+
+#include <cne_mmap.h>
+#include <string.h>
+/* Prototype required to fix "no previous prototype for function" error */
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size);
+
+/* When launching the mmap_alloc fuzz test, the -rss_limit_mb=N parameter
+ * needs to be passed. Otherwise, an out of memory error will happen.
+ */
+extern "C" int
+LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+    uint32_t bufcnt, bufsz;
+    mmap_type_t hugepage;
+    mmap_t *mm;
+
+    if (size > sizeof(bufsz))
+        size = sizeof(bufsz);
+    memcpy(&bufcnt, data, size);
+    hugepage = (mmap_type_t)(bufcnt % MMAP_HUGEPAGE_CNT);
+    mm       = mmap_alloc(bufcnt, size, hugepage);
+    mmap_addr_at_offset(mm, size);
```

Comment:
Should we check the return value of mmap_addr_at_offset() to be valid and within the range of the mapped memory? Also using size as the offset will always give us 0-4 bytes offset. Maybe better to use some value from 0 - (bufcnt * size) and in the code now size can never be larger than 4. I do not think this is what you had in mind maybe??